sentence_transformers/cross_encoder/losses/ListMLELoss.py

```diff
@@ -0,0 +1,313 @@
+from __future__ import annotations
+
+import torch
+from torch import Tensor, nn
+
+from sentence_transformers.cross_encoder import CrossEncoder
+from sentence_transformers.util import fullname
+
+
+class ListMLELambdaWeight(nn.Module):
+    """Base class for implementing weighting schemes in Position-Aware ListMLE Loss."""
+
+    def __init__(self, rank_discount_fn=None) -> None:
+        """
+        Initialize a lambda weight for ListMLE loss.
+
+        Args:
+            rank_discount_fn: Function that computes a discount for each rank position.
+                              If None, uses default discount of 2^(list_size - rank) - 1.
+        """
+        super().__init__()
+        self.rank_discount_fn = rank_discount_fn
+
+    def forward(self, ranks: Tensor, list_size: int) -> Tensor:
+        """
+        Calculate position-aware weights for the ListMLE loss.
+
+        Args:
+            ranks: A tensor of rank positions [batch_size, list_size]
+            list_size: Size of the list
+
+        Returns:
+            Tensor: Weights for each position [batch_size, list_size]
+        """
+        if self.rank_discount_fn is None:
+            # Default rank discount: 2^(list_size - rank) - 1
+            return torch.pow(2.0, list_size - ranks) - 1.0
+        else:
+            return self.rank_discount_fn(ranks)
```

Comment:
```suggestion
        if self.rank_discount_fn is not None:
            return self.rank_discount_fn(ranks)

        # Default rank discount: 2^(list_size - rank) - 1
        return torch.pow(2.0, list_size - ranks) - 1.0
```
Doesn't really matter much, but I prefer this format